sycl/include/sycl/ext/oneapi/experimental/syclbin_kernel_bundle.hpp

```diff
@@ -24,6 +25,39 @@
 
 namespace sycl {
 inline namespace _V1 {
+
+namespace detail {
+__SYCL_EXPORT std::shared_ptr<detail::kernel_bundle_impl>
+link_impl(const kernel_bundle<bundle_state::object> *ObjectBundles,
+          size_t NumObjectBundles, const std::vector<device> &Devs,
+          bool FastLink);
+
+template <
+    typename PropertyListT = ext::oneapi::experimental::empty_properties_t,
+    typename = std::enable_if_t<
+        ext::oneapi::experimental::detail::all_are_properties_of_v<
+            ext::oneapi::experimental::detail::link_props, PropertyListT>>>
+kernel_bundle<bundle_state::executable>
+link_common(const kernel_bundle<bundle_state::object> *ObjectBundles,
+            size_t NumObjectBundles, const std::vector<device> &Devs,
```

Comment:
Do we need to throw an exception if size is zero?

Reply:
Not to my knowledge. The SYCL 2020 specification doesn't seem to mandate such an exception for the regular linking case.